core/shared/src/main/scala/scalaz/zio/stream/Stream.scala

```diff
@@ -303,23 +303,23 @@ trait Stream[+E, +A] { self =>
                         IO.now(Right((rstate, cont, f)))
                       }
                   }
-                  .onError(c => result.done(IO.fail0(c)).void)
+                  .onError(c => result.done(IO.fail(c)).void)
                   .fork
         _ <- fiber.observe.flatMap {
-              case ExitResult.Succeeded(Left(_)) =>
+              case Exit.Success(Left(_)) =>
                 done.done(
-                  IO.terminate(new Exception("Logic error: Stream.peel's inner stream ended with a Left"))
+                  IO.raiseUnchecked(new Exception("Logic error: Stream.peel's inner stream ended with a Left"))
                 )
-              case ExitResult.Succeeded(Right((rstate, _, _))) => done.complete(rstate)
-              case ExitResult.Failed(c)                        => done.done(IO.fail0(c))
+              case Exit.Success(Right((rstate, _, _))) => done.succeed(rstate)
+              case Exit.Failure(c)                     => done.done(IO.fail(c))
```

Comment:
`done.fail(c )`?

Reply:
that doesn't work out

Reply:
you know what, I implemented it -- it's just one line :D